src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -0,0 +1,208 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    /// <summary>Provides an abstraction for time.</summary>
+    public abstract class TimeProvider
+    {
+        private readonly double _tickFrequency;
+
+        /// <summary>
+        /// Gets a <see cref="TimeProvider"/> that provides a clock based on <see cref="DateTimeOffset.UtcNow"/>,
+        /// a time zone based on <see cref="TimeZoneInfo.Local"/>, a high-performance time stamp based on <see cref="Stopwatch"/>,
+        /// and a timer based on <see cref="Timer"/>.
+        /// </summary>
+        public static TimeProvider System { get; } = new SystemTimeProvider(null);
+
+        /// <summary>
+        /// Initializes the instance with the timestamp frequency.
+        /// </summary>
+        /// <param name="timestampFrequency">Frequency of the values returned from <see cref="GetTimestamp"/> method. </param>
```

Comment:
We should add an exception tag here for the ArgumentOutOfRangeException that will be thrown if timestampFrequency <= 0.